pipeline_tools/shared/submission/create_analysis_metadata.py

```diff
@@ -113,6 +119,16 @@ def create_analysis_protocol(
         'computational_method': method,
         'type': get_analysis_protocol_type(),
     }
+
+    string_to_hash = json.dumps(analysis_protocol, sort_keys=True)
+    entity_id = str(uuid.uuid5(NAMESPACE, string_to_hash)).lower()
```

Comment:
Just curious, why use UUID5 here not UUID4? Is this a new spec for the entity IDs?

Reply:
Exactly right - in the new spec, the UUIDs need to be deterministic.